solana/programs/example-native-token-transfers/src/instructions/redeem.rs

```diff
@@ -24,6 +24,8 @@ pub struct Redeem<'info> {
     // NOTE: this works when the contract is paused
     #[account(
         constraint = config.threshold > 0 @ NTTError::ZeroThreshold,
+        constraint = config.next_transceiver_id != 0 @ NTTError::NoRegisteredTransceivers,
```

Comment:
these seem redudnant because we already have a constraint on `transceiver` (line 51) that it must be enabled. So if that constraint passes, it means that it's enabled, so we can't have 0 enabled transceivers

Reply:
I had the same impression, but assumed it was intentional... "to be safe". I disagree with this "to be safe" philosophy, but I assumed that this is what's going on here because of other instances in the past.